packages/kbn-esql-validation-autocomplete/src/autocomplete/__tests__/autocomplete.suggest.eval.test.ts

```diff
@@ -549,5 +549,104 @@ describe('autocomplete.suggest', () => {
         { triggerCharacter: ' ' }
       );
     });
+
+    test('case', async () => {
+      const { assertSuggestions } = await setup();
+      const comparisonOperators = ['==', '!=', '>', '<', '>=', '<=']
+        .map((op) => `${op} `)
+        .concat(',');
+
+      // case( / ) suggest any field/eval function in this position as first argument
+      await assertSuggestions(
+        'from a | eval case(/)',
+        [
+          // With extra space after field name to open suggestions
+          ...getFieldNamesByType('any').map((field) => `${field} `),
+          ...getFunctionSignaturesByReturnType('eval', 'any', { scalar: true }, undefined, [
+            'case',
+          ]),
+        ],
+        {
+          triggerCharacter: ' ',
+        }
+      );
+
+      // case( field /) suggest comparison operators at this point to converge to a boolean
+      await assertSuggestions('from a | eval case( textField /)', comparisonOperators, {
+        triggerCharacter: ' ',
+      });
+      await assertSuggestions('from a | eval case( doubleField /)', comparisonOperators, {
+        triggerCharacter: ' ',
+      });
+      await assertSuggestions('from a | eval case( booleanField /)', comparisonOperators, {
+        triggerCharacter: ' ',
+      });
+
+      // case( field > /) suggest field/function of the same type of the right hand side to complete the boolean expression
+      await assertSuggestions(
+        'from a | eval case( keywordField != /)',
+        [
+          // Notice no extra space after field name
+          ...getFieldNamesByType(['keyword', 'text', 'boolean']).map((field) => `${field}`),
+          ...getFunctionSignaturesByReturnType(
+            'eval',
+            ['keyword', 'text', 'boolean'],
+            { scalar: true },
+            undefined,
+            []
+          ),
+        ],
+        {
+          triggerCharacter: ' ',
+        }
+      );
+      await assertSuggestions(
+        'from a | eval case( integerField != /)',
+        [
+          // Notice no extra space after field name
+          ...getFieldNamesByType(ESQL_COMMON_NUMERIC_TYPES).map((field) => `${field}`),
+          ...getFunctionSignaturesByReturnType(
+            'eval',
+            ESQL_COMMON_NUMERIC_TYPES,
+            { scalar: true },
+            undefined,
+            []
+          ),
+        ],
+        {
+          triggerCharacter: ' ',
+        }
+      );
```

Comment:
let's add a test for this without a trigger character as well :)

Reply:
Added here [`9c6002b` (#192135)](https://github.com/elastic/kibana/pull/192135/commits/9c6002b5c959a9f88f54e40c7e516fd33e74bafb)